source/draw/gpu/GpuCanvas.ooc

```diff
@@ -98,5 +98,25 @@ GpuCanvas: abstract class extends Canvas {
 		translation := FloatTransform3D createTranslation(normalizedBox leftTop x, normalizedBox leftTop y, 0.0f)
 		translation * scaling
 	}
+	_createView: func (targetSize: FloatVector2D, normalizedView: FloatTransform3D) -> FloatTransform3D {
+		this _toLocal * normalizedView normalizedToReference(targetSize) * this _toLocal
+	}
+	_createProjection: func (targetSize: FloatVector2D, focalLengthPerWidth: Float) -> FloatTransform3D {
+		result: FloatTransform3D
+		focalLengthPerHeight := focalLengthPerWidth * (targetSize x as Float) / (targetSize y as Float)
+		flipX := this _coordinateTransform a as Float
+		flipY := -(this _coordinateTransform e as Float)
+		if (focalLengthPerWidth > 0.0f) {
+			nearPlane := 0.01f
+			farPlane := 10000.0f
+			a := flipX * 2.0f * focalLengthPerWidth
+			f := flipY * 2.0f * focalLengthPerHeight
+			k := (farPlane + nearPlane) / (farPlane - nearPlane)
+			o := 2.0f * farPlane * nearPlane / (farPlane - nearPlane)
+			result = FloatTransform3D new(a, 0.0f, 0.0f, 0.0f, 0.0f, f, 0.0f, 0.0f, 0.0f, 0.0f, k, -1.0f, 0.0f, 0.0f, o, 0.0f)
+		} else
+			result = FloatTransform3D createScaling(flipX * 2.0f / targetSize x, flipY * 2.0f / targetSize y, 0.0f)
+		result
+	}
```

Comment:
This was moved here from OpenGLCanvas so that UnixWindow could use it too and get rid of dependencies to obsoleted variables.